plugins/modules/aci_fabric_wide_settings.py

```diff
@@ -0,0 +1,313 @@
+  returned: info
+  type: list
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        disable_remote_ep_learning=dict(type="bool"),
+        enforce_subnet_check=dict(type="bool"),
+        enforce_epg_vlan_validation=dict(type="bool"),
+        enforce_domain_validation=dict(type="bool"),
+        spine_opflex_client_auth=dict(type="bool"),
+        leaf_opflex_client_auth=dict(type="bool"),
+        spine_ssl_opflex=dict(type="bool"),
+        leaf_ssl_opflex=dict(type="bool"),
+        ssl_opflex_tls_10=dict(type="bool"),
+        ssl_opflex_tls_11=dict(type="bool"),
+        ssl_opflex_tls_12=dict(type="bool"),
+        reallocate_gipo=dict(type="bool"),
+        restrict_infra_vlan_traffic=dict(type="bool"),
+        state=dict(type="str", default="present", choices=["present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_together=[["ssl_opflex_tls_10", "ssl_opflex_tls_11", "ssl_opflex_tls_12"]],
```

Comment:
Is required together needed?

If so i would suggest a few possible improvements:
- we could create an ssl opflex dict with the 3 inputs, and put these sub options as required then the required_together is not needed and you conditional is easier further in the code.
- we could add more clear documentation that these linked together

Reply:
These are used together to construct the "opflexpSslProtocols" parameter, which is a comma separated list of the TLS versions. The reason I added them in a required_together is to avoid a situation where someone adds a TLS version but in doing so inadvertently removes the existing TLS versions.